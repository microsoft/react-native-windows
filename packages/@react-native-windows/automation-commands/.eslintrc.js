module.exports = {
    extends: ['@rnw-scripts'],
    parserOptions: {tsconfigRootDir : __dirname},
};
