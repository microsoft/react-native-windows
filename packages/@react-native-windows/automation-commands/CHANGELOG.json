{
  "name": "@react-native-windows/automation-commands",
  "entries": [
    {
      "date": "Fri, 01 Oct 2021 05:05:54 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-2",
      "version": "0.0.6-2",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-2",
            "author": "julio.rocha@microsoft.com",
            "commit": "eaefea114554d0b91dbf2d6aff650b2fc5673f7a",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 30 Sep 2021 05:06:48 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-1",
      "version": "0.0.6-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-1",
            "author": "erozell@outlook.com",
            "commit": "4bc8637395ea828788c0ba267561c1a01f06f484",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 29 Sep 2021 05:06:54 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-0",
      "version": "0.0.6-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-0",
            "author": "anandrag@microsoft.com",
            "commit": "eb04aeb2e5fcb74988f1630f24adc27a034351fe",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 28 Sep 2021 05:07:03 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.5",
      "version": "0.0.5",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.4",
            "author": "34109996+chiaramooney@users.noreply.github.com",
            "commit": "d4970fdd583866c3ac7af6a0cf38040cf3b22d93",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 25 Sep 2021 05:06:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.5-0",
      "version": "0.0.5-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-12",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "eaf932469678b8b67b0b2b17fd2bcdda1ed4e358",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 24 Sep 2021 05:06:42 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4",
      "version": "0.0.4",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-11",
            "author": "igklemen@microsoft.com",
            "commit": "8b63ada9ba1d55bfd161388fc18c2b5c6445b508",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 23 Sep 2021 05:06:48 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4-1",
      "version": "0.0.4-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-10",
            "author": "34109996+chiaramooney@users.noreply.github.com",
            "commit": "3813907b2eaa6198f0a8b9dc570075f0c7117301",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 22 Sep 2021 05:06:43 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4-0",
      "version": "0.0.4-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-9",
            "author": "ali-hk@users.noreply.github.com",
            "commit": "0bc36a30ab3cd5bb4cfd767e503b29c185cfcddd",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 21 Sep 2021 05:08:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3",
      "version": "0.0.3",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-8",
            "author": "igklemen@microsoft.com",
            "commit": "08eb19f42f4f74bd82f33db2e27ba2d33b2ddd1b",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 18 Sep 2021 05:07:45 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-7",
      "version": "0.0.3-7",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-7",
            "author": "erozell@outlook.com",
            "commit": "fbbfb83843c60abb8c02368077fd0a406f0bdcc0",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 17 Sep 2021 05:06:53 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-6",
      "version": "0.0.3-6",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-6",
            "author": "jthysell@microsoft.com",
            "commit": "95fecc02d3dcc9f03d0fc7d47d30cb45e128bca6",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 16 Sep 2021 05:06:44 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-5",
      "version": "0.0.3-5",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-5",
            "author": "erozell@outlook.com",
            "commit": "b46965a75f179c1855a2499c7f3c6b4b10daf416",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 15 Sep 2021 05:07:04 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-4",
      "version": "0.0.3-4",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-4",
            "author": "30809111+acoates-ms@users.noreply.github.com",
            "commit": "5dbd74b1852c38f851483478d6da7d81949d0195",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 14 Sep 2021 05:07:02 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-3",
      "version": "0.0.3-3",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-3",
            "author": "jthysell@microsoft.com",
            "commit": "d3de4a2b04415ff9004b7ba08ee21787b924d0d4",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 11 Sep 2021 05:06:55 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-2",
      "version": "0.0.3-2",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-2",
            "author": "anandrag@microsoft.com",
            "commit": "09bc9974c8721bf5559680dc7a8aa00514c1c7f3",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 10 Sep 2021 05:07:31 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-1",
      "version": "0.0.3-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-1",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "42d0853c684f0e9f9c9f1f7fa34cbf4ca3810edc",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 09 Sep 2021 05:07:46 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-0",
      "version": "0.0.3-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-0",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "66a9f86f24f862c9e9e611ad435f68daddb4e04e",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 08 Sep 2021 05:08:53 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.2",
      "version": "0.0.2",
      "comments": {
        "patch": [
          {
            "comment": "Set consistent node requirements on our packages",
            "author": "ngerlem@microsoft.com",
            "commit": "2974ea0ab58b546264b8d9a4a7c12ceeb0a02851",
            "package": "@react-native-windows/automation-commands"
          }
        ],
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.2",
            "author": "ngerlem@microsoft.com",
            "commit": "2974ea0ab58b546264b8d9a4a7c12ceeb0a02851",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 04 Sep 2021 05:06:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.2-0",
      "version": "0.0.2-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.2-0",
            "author": "dannyvv@microsoft.com",
            "commit": "d0c3465f27bd0de739b23b46e957cbdab6bdf014",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 03 Sep 2021 05:06:04 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.1",
      "version": "0.0.1",
      "comments": {
        "patch": [
          {
            "comment": "@react-native-windows/automation",
            "author": "ngerlem@microsoft.com",
            "commit": "17e32d36180fa5c7e45c3c926c6eaa729efb6ea0",
            "package": "@react-native-windows/automation-commands"
          }
        ],
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.1",
            "author": "erozell@outlook.com",
            "commit": "439b8a54b62807068bfa17e130f8ab0378eaaf36",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    }
  ]
}
