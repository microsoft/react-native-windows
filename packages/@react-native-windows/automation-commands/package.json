{
  "name": "@react-native-windows/automation-commands",
  "version": "0.1.31",
  "description": "Allows controlling your react-native-windows application",
  "main": "lib-commonjs/index.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git@github.com:microsoft/react-native-windows.git",
    "directory": "packages/@react-native-windows/automation-commands"
  },
  "private": true,
  "scripts": {
    "build": "rnw-scripts build",
    "clean": "rnw-scripts clean",
    "lint": "rnw-scripts lint",
    "lint:fix": "rnw-scripts lint:fix",
    "watch": "rnw-scripts watch"
  },
  "dependencies": {
    "@react-native-windows/automation-channel": "^0.5.11",
    "@typescript-eslint/eslint-plugin": "^5.30.5",
    "@typescript-eslint/parser": "^5.30.5"
  },
  "devDependencies": {
    "@jest/types": "^29.2.1",
    "@rnw-scripts/eslint-config": "1.1.14",
    "@rnw-scripts/just-task": "2.3.3",
    "@rnw-scripts/ts-config": "2.0.2",
    "@types/jest": "^29.2.2",
    "@types/node": "^14.14.22",
    "eslint": "^8.19.0",
    "prettier": "^2.4.1",
    "typescript": "^4.9.5"
  },
  "files": [
    "lib-commonjs",
    "README.md"
  ],
  "engines": {
    "node": ">= 14"
  }
}