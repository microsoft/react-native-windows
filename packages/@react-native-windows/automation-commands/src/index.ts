/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import dumpVisualTree from './dumpVisualTree';

export {dumpVisualTree};
