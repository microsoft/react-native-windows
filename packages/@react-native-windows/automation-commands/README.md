# @react-native-windows/automation-commands

`@react-native-windows/automation-commands` provides built-in commands that may
be used with `@react-native-windows/automation-channel` to make it easier to
test your application.

**This package is a work in progress**

## Adding to your application

**TBD**

## Commands

### dumpVisualTree

The `dumpVisualTree` command creates a JSON object corresponding to the XAML
tree under a given testID. This can be used in conjunction with snapshot
testing to validate that your native UI looks how you could expect.

```ts
import {dumpVisualTree} from '@react-native-windows/automation-commands';

test('Widget', async () => {
  const dump = await dumpVisualTree('widget-test-id');
  expect(dump).toMatchSnapshot();
});
```

#### Options
**TBD**