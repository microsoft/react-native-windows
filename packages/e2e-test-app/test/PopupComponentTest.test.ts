import {dumpVisualTree} from '@react-native-windows/automation-commands';
import {goToComponentExample} from './RNTesterNavigation';
import {verifyNoErrorLogs} from './Helpers';

beforeAll(async () => {
  await goToComponentExample('Popup');
});

afterEach(async () => {
  await verifyNoErrorLogs();
});

describe('PopupTests', () => {
  test('Popups can be anchored', async () => {
    const dump = await dumpVisualTree('anchor-popup');
    expect(dump).toMatchSnapshot();
  });
  test('Popups can have different placement', async () => {
    const dump = await dumpVisualTree('popup-placement');
    expect(dump).toMatchSnapshot();
  });
  test('Popups can have customized accessibility', async () => {
    const dump = await dumpVisualTree('popup-accessibility');
    expect(dump).toMatchSnapshot();
  });
});