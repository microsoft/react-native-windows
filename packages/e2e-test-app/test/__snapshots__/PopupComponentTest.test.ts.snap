// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupTests Popups can be anchored 1`] = `
Object {
  "AutomationId": "anchor-popup",
  "Background": null,
  "BorderBrush": "#00000000",
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Height": 189,
  "HorizontalAlignment": "Stretch",
  "Left": 0,
  "Margin": "0,0,0,0",
  "Top": 0,
  "VerticalAlignment": "Stretch",
  "Visibility": "Visible",
  "Width": 718,
  "XamlType": "Microsoft.ReactNative.ViewPanel",
  "children": Array [
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "The following tests popup Anchor",
      "Top": 0,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 250,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Background": null,
      "BorderBrush": "#00000000",
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Height": 32,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Top": 19,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 718,
      "XamlType": "Microsoft.ReactNative.ViewPanel",
      "children": Array [
        Object {
          "Clip": null,
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "Height": 32,
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "10,10,10,10",
          "Text": "Text Input to Anchor popup to: ",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 300,
          "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
        },
        Object {
          "Background": "#B3FFFFFF",
          "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
          "BorderThickness": "1,1,1,1",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Foreground": "#E4000000",
          "Height": 32,
          "HorizontalAlignment": "Stretch",
          "Left": 300,
          "Margin": "0,0,0,0",
          "Padding": "10,5,6,6",
          "Text": "",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 300,
          "XamlType": "Windows.UI.Xaml.Controls.TextBox",
          "children": Array [
            Object {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Grid",
              "children": Array [
                Object {
                  "Background": "#B3FFFFFF",
                  "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                  "BorderThickness": "1,1,1,1",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "BorderElement",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                },
                Object {
                  "Background": "#00FFFFFF",
                  "BorderBrush": "#00FFFFFF",
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#E4000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "1,1,1,1",
                  "Name": "ContentElement",
                  "Padding": "10,5,6,6",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ScrollViewer",
                  "children": Array [
                    Object {
                      "Background": null,
                      "BorderBrush": "#00FFFFFF",
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "Root",
                      "Padding": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Controls.Border",
                      "children": Array [
                        Object {
                          "Background": "#00FFFFFF",
                          "BorderBrush": null,
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "HorizontalAlignment": "Stretch",
                          "Left": 0,
                          "Margin": "0,0,0,0",
                          "Padding": "0,0,0,0",
                          "Top": 0,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "XamlType": "Windows.UI.Xaml.Controls.Grid",
                          "children": Array [
                            Object {
                              "Background": null,
                              "BorderBrush": null,
                              "BorderThickness": "0,0,0,0",
                              "Clip": "Windows.UI.Xaml.Media.RectangleGeometry",
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "Foreground": "#E4000000",
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "10,5,6,6",
                              "Name": "ScrollContentPresenter",
                              "Padding": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "XamlType": "Windows.UI.Xaml.Controls.ScrollContentPresenter",
                              "children": Array [
                                Object {
                                  "Clip": null,
                                  "FlowDirection": "LeftToRight",
                                  "HorizontalAlignment": "Stretch",
                                  "Left": 0,
                                  "Margin": "0,0,0,0",
                                  "Top": 0,
                                  "VerticalAlignment": "Stretch",
                                  "Visibility": "Visible",
                                  "XamlType": "Windows.UI.Xaml.FrameworkElement",
                                },
                              ],
                            },
                            Object {
                              "Background": null,
                              "BorderBrush": null,
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "0,0,0,0",
                              "Padding": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "XamlType": "Windows.UI.Xaml.Controls.Grid",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#99000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "DescriptionPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
              ],
            },
          ],
        },
      ],
    },
    Object {
      "Background": null,
      "BorderBrush": "#00000000",
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Height": 138,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Top": 51,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 718,
      "XamlType": "Microsoft.ReactNative.ViewPanel",
      "children": Array [
        Object {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "Height": 38,
          "HorizontalAlignment": "Stretch",
          "Left": 50,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 50,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 618,
          "XamlType": "Microsoft.ReactNative.ViewControl",
          "children": Array [
            Object {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Foreground": "#FF000000",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
              "children": Array [
                Object {
                  "Background": "#B3FFFFFF",
                  "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                  "BorderThickness": "1,1,1,1.5",
                  "Clip": null,
                  "CornerRadius": "3,3,3,3",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                  "children": Array [
                    Object {
                      "Background": null,
                      "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                      "BorderThickness": "1,1,1,1.5",
                      "Clip": null,
                      "CornerRadius": "3,3,3,3",
                      "FlowDirection": "LeftToRight",
                      "Height": 38,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 618,
                      "XamlType": "Microsoft.ReactNative.ViewPanel",
                      "children": Array [
                        Object {
                          "Background": null,
                          "BorderBrush": "#00000000",
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "Height": 35,
                          "HorizontalAlignment": "Stretch",
                          "Left": 1,
                          "Margin": "0,0,0,0",
                          "Top": 1,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Width": 616,
                          "XamlType": "Microsoft.ReactNative.ViewPanel",
                          "children": Array [
                            Object {
                              "Background": null,
                              "BorderBrush": "#00000000",
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "Height": 35,
                              "HorizontalAlignment": "Stretch",
                              "Left": 0,
                              "Margin": "0,0,0,0",
                              "Top": 0,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Width": 616,
                              "XamlType": "Microsoft.ReactNative.ViewPanel",
                              "children": Array [
                                Object {
                                  "Clip": null,
                                  "FlowDirection": "LeftToRight",
                                  "Foreground": "#E4000000",
                                  "Height": 19,
                                  "HorizontalAlignment": "Stretch",
                                  "Left": 8,
                                  "Margin": "0,0,0,0",
                                  "Padding": "0,0,0,0",
                                  "Text": "Open Popup",
                                  "Top": 8,
                                  "VerticalAlignment": "Stretch",
                                  "Visibility": "Visible",
                                  "Width": 600,
                                  "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                                },
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
  ],
}
`;

exports[`PopupTests Popups can have different placement 1`] = `
Object {
  "AutomationId": "popup-placement",
  "Background": null,
  "BorderBrush": "#00000000",
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Height": 500,
  "HorizontalAlignment": "Stretch",
  "Left": 0,
  "Margin": "0,0,0,0",
  "Top": 0,
  "VerticalAlignment": "Stretch",
  "Visibility": "Visible",
  "Width": 500,
  "XamlType": "Microsoft.ReactNative.ViewPanel",
  "children": Array [
    Object {
      "Background": null,
      "BorderBrush": null,
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 37,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Top": 0,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 500,
      "XamlType": "Microsoft.ReactNative.ViewControl",
      "children": Array [
        Object {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Foreground": "#FF000000",
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
          "children": Array [
            Object {
              "Background": "#B3FFFFFF",
              "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
              "BorderThickness": "1,1,1,1.5",
              "Clip": null,
              "CornerRadius": "3,3,3,3",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Stretch",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Border",
              "children": Array [
                Object {
                  "Background": null,
                  "BorderBrush": "Windows.UI.Xaml.Media.LinearGradientBrush",
                  "BorderThickness": "1,1,1,1.5",
                  "Clip": null,
                  "CornerRadius": "3,3,3,3",
                  "FlowDirection": "LeftToRight",
                  "Height": 37,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 500,
                  "XamlType": "Microsoft.ReactNative.ViewPanel",
                  "children": Array [
                    Object {
                      "Background": null,
                      "BorderBrush": "#00000000",
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "Height": 35,
                      "HorizontalAlignment": "Stretch",
                      "Left": 1,
                      "Margin": "0,0,0,0",
                      "Top": 1,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 498,
                      "XamlType": "Microsoft.ReactNative.ViewPanel",
                      "children": Array [
                        Object {
                          "Background": null,
                          "BorderBrush": "#00000000",
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "Height": 35,
                          "HorizontalAlignment": "Stretch",
                          "Left": 0,
                          "Margin": "0,0,0,0",
                          "Top": 0,
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Width": 498,
                          "XamlType": "Microsoft.ReactNative.ViewPanel",
                          "children": Array [
                            Object {
                              "Clip": null,
                              "FlowDirection": "LeftToRight",
                              "Foreground": "#E4000000",
                              "Height": 19,
                              "HorizontalAlignment": "Stretch",
                              "Left": 8,
                              "Margin": "0,0,0,0",
                              "Padding": "0,0,0,0",
                              "Text": "Toggle popup",
                              "Top": 8,
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Width": 482,
                              "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
  ],
}
`;
